crates/supervisor/storage/src/log/mdbx.rs

```diff
@@ -0,0 +1,311 @@
+//! Reth's MDBX-backed abstraction of [`LogStorage`](crate::log::LogStorage) for superchain state.
+//!
+//! This module provides the [`MdbxLogStorage`] struct, which uses the
+//! [`reth-db`] abstraction of reth to store execution logs
+//! and block metadata required by the Optimism supervisor.
+//!
+//! It supports:
+//! - Writing full blocks of logs with metadata
+//! - Retrieving block metadata by number
+//! - Finding a block from a specific log (with hash/index match)
+//! - Fetching logs per block using dup-sorted key layout
+//!
+//! Logs are stored in [`LogEntries`] under dup-sorted tables, with log index
+//! used as the subkey. Block metadata is stored in [`BlockRefs`].
+
+use crate::{
+    error::StorageError,
+    log::{LogStorageReader, LogStorageWriter},
+    models::{BlockRefs, LogEntries},
+};
+use kona_protocol::BlockInfo;
+use kona_supervisor_types::Log;
+use reth_db_api::{
+    cursor::{DbCursorRO, DbDupCursorRO, DbDupCursorRW},
+    transaction::{DbTx, DbTxMut},
+};
+use std::fmt::Debug;
+use tracing::{debug, error};
+
+/// A log storage that wraps a transactional reference to the MDBX backend.
+#[derive(Debug)]
+#[allow(dead_code)]
+pub(crate) struct MdbxLogStorage<'tx, TX> {
+    tx: &'tx TX,
+}
+
+/// Internal constructor and setup methods for [`MdbxLogStorage`].
+impl<'tx, TX> MdbxLogStorage<'tx, TX> {
+    #[allow(dead_code)]
+    pub(crate) const fn new(tx: &'tx TX) -> Self {
+        Self { tx }
+    }
+}
+
+/// Implements the [`LogStorageWriter`] trait
+impl<TX> LogStorageWriter for MdbxLogStorage<'_, TX>
+where
+    TX: DbTxMut,
+{
+    type Error = StorageError;
+
+    fn store_block_logs(&self, block: &BlockInfo, logs: Vec<Log>) -> Result<(), Self::Error> {
+        debug!(target: "supervisor_storage", "Storing {} logs for block {}", logs.len(), block.number);
+        let tx = self.tx;
+        if let Err(e) = tx.put::<BlockRefs>(block.number, (*block).into()) {
+            error!(target: "supervisor_storage", "Failed to write block for {}: {e}", block.number);
```

Comment:
I think we should do structured logging. 
```suggestion
            error!(
                target: "supervisor_storage",
                block_number = block.number,
                error = %e,
                "Failed to write block"
            );
```

Reply:
may be should go with full error trace? like `?e`